src/txn/txn_ckpt.c

```diff
@@ -1214,30 +1202,51 @@ __checkpoint_lock_dirty_tree_int(WT_SESSION_IMPL *session, bool is_checkpoint, b
 {
     WT_CONNECTION_IMPL *conn;
     WT_DECL_RET;
+    u_int max_ckpt_drop;
+    bool is_wt_ckpt;
 
     WT_UNUSED(is_checkpoint);
     conn = S2C(session);
 
-    /*
-     * We can't delete checkpoints if a backup cursor is open. WiredTiger checkpoints are uniquely
-     * named and it's OK to have multiple of them in the system: clear the delete flag for them, and
-     * otherwise fail. Hold the lock until we're done (blocking hot backups from starting), we don't
-     * want to race with a future hot backup.
-     */
-    if (conn->hot_backup)
-        WT_CKPT_FOREACH (ckptbase, ckpt) {
-            if (!F_ISSET(ckpt, WT_CKPT_DELETE))
-                continue;
-            if (WT_PREFIX_MATCH(ckpt->name, WT_CHECKPOINT)) {
+    /* Check that it is OK to remove all the checkpoints marked for deletion. */
+    max_ckpt_drop = 0;
+    WT_CKPT_FOREACH (ckptbase, ckpt) {
+        if (!F_ISSET(ckpt, WT_CKPT_DELETE))
+            continue;
+        is_wt_ckpt = WT_PREFIX_MATCH(ckpt->name, WT_CHECKPOINT);
+
+/*
+ * If there is a hot backup, don't delete any WiredTiger checkpoint that could possibly have been
+ * created before the backup started. Fail if trying to delete any other named checkpoint.
+ */
+#ifdef DISABLED_CODE
+        if (conn->hot_backup_start != 0 && ckpt->sec <= conn->hot_backup_start) {
+#else
```

Comment:
This `#ifdef` was part of the temporary patch in WT-6141, and was removed in WT-6118. You should keep the `if` statement from the `#ifdef DISABLED_CODE` branch and eliminate the `#else` branch.

Reply:
Done.